// frontend/src/mockData.js
// Mock data for Genius36 Technologies website

export const heroImages = {
  primary: "https://images.unsplash.com/photo-1644088379091-d574269d422f?crop=entropy&cs=srgb&fm=jpg&ixid=M3w3NDQ2NDF8MHwxfHNlYXJjaHwxfHx0ZWNobm9sb2d5JTIwYmFja2dyb3VuZHxlbnwwfHx8fDE3NjY4MjIyMDF8MA&ixlib=rb-4.1.0&q=85",
  secondary: "https://images.pexels.com/photos/5716032/pexels-photo-5716032.jpeg",
  abstract: "https://images.unsplash.com/photo-1579548122080-c35fd6820ecb?crop=entropy&cs=srgb&fm=jpg&ixid=M3w3NDQ2NDF8MHwxfHNlYXJjaHwzfHx0ZWNobm9sb2d5JTIwYmFja2dyb3VuZHxlbnwwfHx8fDE3NjY4MjIyMDF8MA&ixlib=rb-4.1.0&q=85"
};

export const servicesImages = {
  general: "https://images.pexels.com/photos/845451/pexels-photo-845451.jpeg",
  networking: "https://images.pexels.com/photos/1054397/pexels-photo-1054397.jpeg",
  development: "https://images.unsplash.com/photo-1610563166150-b34df4f3bcd6?crop=entropy&cs=srgb&fm=jpg&ixid=M3w3NDk1ODF8MHwxfHNlYXJjaHwyfHxzb2Z0d2FyZSUyMGRldmVsb3BtZW50fGVufDB8fHx8MTc2NjcyOTYxM3ww&ixlib=rb-4.1.0&q=85",
  team: "https://images.unsplash.com/photo-1522071820081-009f0129c71c?crop=entropy&cs=srgb&fm=jpg&ixid=M3w3NDk1ODF8MHwxfHNlYXJjaHwzfHxzb2Z0d2FyZSUyMGRldmVsb3BtZW50fGVufDB8fHx8MTc2NjcyOTYxM3ww&ixlib=rb-4.1.0&q=85",
  workspace: "https://images.unsplash.com/photo-1579403124614-197f69d8187b?crop=entropy&cs=srgb&fm=jpg&ixid=M3w3NDk1ODF8MHwxfHNlYXJjaHw0fHxzb2Z0d2FyZSUyMGRldmVsb3BtZW50fGVufDB8fHx8MTc2NjcyOTYxM3ww&ixlib=rb-4.1.0&q=85",
  mobile: "https://images.pexels.com/photos/1181244/pexels-photo-1181244.jpeg"
};

export const trainingImages = {
  programming: "https://images.pexels.com/photos/249798/pexels-photo-249798.png",
  classroom: "https://images.pexels.com/photos/3747481/pexels-photo-3747481.jpeg",
  session: "https://images.pexels.com/photos/33920044/pexels-photo-33920044.jpeg",
  learning: "https://images.pexels.com/photos/1181474/pexels-photo-1181474.jpeg"
};

export const aboutImages = {
  teamCollaboration: "https://images.unsplash.com/photo-1522071820081-009f0129c71c?crop=entropy&cs=srgb&fm=jpg&ixid=M3w3NDQ2Mzl8MHwxfHNlYXJjaHwxfHx0ZWNobm9sb2d5JTIwdGVhbXxlbnwwfHx8fDE3NjY3Mjk2MDd8MA&ixlib=rb-4.1.0&q=85",
  itProfessionals: "https://images.unsplash.com/photo-1632910121591-29e2484c0259?crop=entropy&cs=srgb&fm=jpg&ixid=M3w3NDQ2Mzl8MHwxfHNlYXJjaHwyfHx0ZWNobm9sb2d5JTIwdGVhbXxlbnwwfHx8fDE3NjY3Mjk2MDd8MA&ixlib=rb-4.1.0&q=85",
  friendlyTeam: "https://images.unsplash.com/photo-1531539427495-97c44a449837?crop=entropy&cs=srgb&fm=jpg&ixid=M3w3NDQ2Mzl8MHwxfHNlYXJjaHwzfHx0ZWNobm9sb2d5JTIwdGVhbXxlbnwwfHx8fDE3NjY3Mjk2MDd8MA&ixlib=rb-4.1.0&q=85",
  office: "https://images.pexels.com/photos/6804068/pexels-photo-6804068.jpeg",
  building: "https://images.pexels.com/photos/323705/pexels-photo-323705.jpeg"
};

export const services = [
  {
    id: 1,
    title: "Custom Website Design & Development",
    description: "Create stunning, responsive websites tailored to your business needs with modern technologies and best practices.",
    icon: "Globe",
    features: ["Responsive Design", "SEO Optimized", "Fast Loading", "Custom CMS"]
  },
  {
    id: 2,
    title: "Software Development",
    description: "Enterprise-grade software solutions for web and mobile platforms with scalable architecture.",
    icon: "Code",
    features: ["Custom Solutions", "Scalable Architecture", "API Integration", "Cloud-Ready"]
  },
  {
    id: 3,
    title: "IT Consulting & Digital Transformation",
    description: "Strategic IT consulting to modernize your business operations and drive digital innovation.",
    icon: "Lightbulb",
    features: ["Strategy Planning", "Process Optimization", "Technology Audit", "Digital Roadmap"]
  },
  {
    id: 4,
    title: "CCTV Camera Installation & Security",
    description: "Professional security camera installation with 24/7 monitoring and advanced surveillance systems.",
    icon: "Video",
    features: ["HD Cameras", "Remote Monitoring", "Night Vision", "Cloud Storage"]
  },
  {
    id: 5,
    title: "Computer Hardware Sales & Repair",
    description: "Quality computer hardware, peripherals, and expert repair services with AMC support.",
    icon: "HardDrive",
    features: ["Genuine Products", "Expert Repair", "AMC Support", "Quick Service"]
  },
  {
    id: 6,
    title: "Networking & IT Infrastructure",
    description: "Complete networking solutions including LAN/WAN setup, server configuration, and infrastructure management.",
    icon: "Network",
    features: ["LAN/WAN Setup", "Server Configuration", "Network Security", "24/7 Support"]
  },
  {
    id: 7,
    title: "Cloud Solutions & Server Setup",
    description: "Scalable cloud infrastructure and server management for modern businesses.",
    icon: "Cloud",
    features: ["Cloud Migration", "Server Management", "Backup Solutions", "High Availability"]
  },
  {
    id: 8,
    title: "IT Training & Programming Courses",
    description: "Industry-aligned IT training programs with hands-on learning and career development.",
    icon: "GraduationCap",
    features: ["Expert Trainers", "Hands-on Learning", "Certification", "Placement Support"]
  },
  {
    id: 9,
    title: "Data Recovery Services",
    description: "Professional data recovery solutions for hard drives, SSDs, RAID systems, and all storage devices.",
    icon: "HardDriveDownload",
    features: ["99% Success Rate", "Secure Process", "All Storage Types", "Free Diagnostics"]
  }
];

export const courses = [
  {
    id: 1,
    title: "Full Stack Web Development",
    duration: "6 Months",
    level: "Beginner to Advanced",
    description: "Master modern web development with React, Node.js, MongoDB, and deployment.",
    modules: ["HTML/CSS/JavaScript", "React.js", "Node.js & Express", "MongoDB", "Git & GitHub", "Deployment"],
    price: "₹45,000",
    image: trainingImages.programming
  },
  {
    id: 2,
    title: "Python Programming & Data Science",
    duration: "4 Months",
    level: "Beginner to Intermediate",
    description: "Learn Python programming, data analysis, machine learning basics, and automation.",
    modules: ["Python Basics", "Data Structures", "Pandas & NumPy", "Data Visualization", "ML Basics", "Automation"],
    price: "₹35,000",
    image: trainingImages.classroom
  },
  {
    id: 3,
    title: "Mobile App Development",
    duration: "5 Months",
    level: "Intermediate",
    description: "Build cross-platform mobile applications using React Native and Flutter.",
    modules: ["React Native", "Flutter", "Firebase", "API Integration", "App Deployment", "UI/UX Design"],
    price: "₹50,000",
    image: trainingImages.session
  },
  {
    id: 4,
    title: "Digital Marketing & SEO",
    duration: "3 Months",
    level: "Beginner",
    description: "Master digital marketing strategies, SEO, social media, and analytics.",
    modules: ["SEO Fundamentals", "Social Media Marketing", "Google Ads", "Content Marketing", "Analytics", "Email Marketing"],
    price: "₹25,000",
    image: trainingImages.learning
  },
  {
    id: 5,
    title: "Cloud Computing & DevOps",
    duration: "4 Months",
    level: "Advanced",
    description: "Learn AWS, Docker, Kubernetes, and modern DevOps practices.",
    modules: ["AWS Fundamentals", "Docker", "Kubernetes", "CI/CD", "Terraform", "Monitoring"],
    price: "₹55,000",
    image: trainingImages.programming
  },
  {
    id: 6,
    title: "Cybersecurity Fundamentals",
    duration: "3 Months",
    level: "Intermediate",
    description: "Understand security principles, ethical hacking, and network security.",
    modules: ["Network Security", "Ethical Hacking", "Cryptography", "Security Tools", "Penetration Testing", "Compliance"],
    price: "₹40,000",
    image: trainingImages.classroom
  }
];

export const projects = [
  {
    id: 1,
    title: "E-Commerce Platform for Retail Chain",
    category: "Web Development",
    client: "Retail Corp",
    description: "Developed a scalable e-commerce platform with inventory management, payment gateway, and real-time analytics.",
    technologies: ["React", "Node.js", "MongoDB", "AWS"],
    image: servicesImages.development,
    year: "2024"
  },
  {
    id: 2,
    title: "Hospital Management System",
    category: "Software Development",
    client: "City Hospital",
    description: "Comprehensive HMS with patient records, appointment scheduling, billing, and pharmacy management.",
    technologies: ["Python", "Django", "PostgreSQL", "React"],
    image: servicesImages.workspace,
    year: "2024"
  },
  {
    id: 3,
    title: "Corporate Network Infrastructure",
    category: "Networking",
    client: "Manufacturing Ltd",
    description: "Designed and deployed complete network infrastructure for 500+ workstations with security protocols.",
    technologies: ["Cisco", "Firewall", "VPN", "Active Directory"],
    image: servicesImages.networking,
    year: "2023"
  },
  {
    id: 4,
    title: "Smart Campus Security System",
    category: "CCTV & Security",
    client: "University Campus",
    description: "Installed 200+ HD CCTV cameras with AI-powered monitoring and cloud storage solution.",
    technologies: ["IP Cameras", "NVR", "Cloud Storage", "AI Analytics"],
    image: servicesImages.general,
    year: "2023"
  },
  {
    id: 5,
    title: "Cloud Migration for Financial Services",
    category: "Cloud Solutions",
    client: "FinTech Company",
    description: "Successfully migrated on-premise infrastructure to AWS with high availability and disaster recovery.",
    technologies: ["AWS", "Docker", "Kubernetes", "Terraform"],
    image: servicesImages.team,
    year: "2024"
  },
  {
    id: 6,
    title: "Mobile Banking Application",
    category: "Mobile Development",
    client: "Regional Bank",
    description: "Feature-rich mobile banking app with biometric authentication and real-time transactions.",
    technologies: ["React Native", "Node.js", "MongoDB", "Firebase"],
    image: servicesImages.mobile,
    year: "2023"
  }
];

export const blogPosts = [
  {
    id: 1,
    title: "Top 10 Cybersecurity Threats in 2025 and How to Protect Your Business",
    excerpt: "Discover the latest cybersecurity challenges facing businesses and learn practical strategies to safeguard your digital assets.",
    author: "Rajesh Kumar",
    date: "2025-02-10",
    category: "Cybersecurity",
    readTime: "8 min read",
    image: servicesImages.networking,
    content: `In today's digital landscape, cybersecurity threats are evolving at an unprecedented pace. As businesses increasingly rely on digital infrastructure, understanding and mitigating these threats has become crucial for survival. This comprehensive guide explores the top 10 cybersecurity threats facing businesses in 2025 and provides actionable strategies to protect your organization.

## Understanding the Current Threat Landscape

The cybersecurity landscape in 2025 is more complex than ever before. Cybercriminals are leveraging advanced technologies like artificial intelligence and machine learning to launch sophisticated attacks. Small businesses are no longer safe targets—they're often preferred because they typically have weaker security measures in place.

The cost of cyber attacks continues to rise, with the average data breach costing organizations millions of dollars. Beyond financial losses, businesses face reputational damage, legal consequences, and operational disruptions that can be devastating.

## Top 10 Cybersecurity Threats in 2025

### 1. Ransomware Attacks

Ransomware remains one of the most significant threats to businesses worldwide. These attacks encrypt your data and demand payment for decryption keys. Modern ransomware attacks are more sophisticated, often targeting backups and using double extortion tactics.

**Protection Strategies:**
- Implement regular automated backups stored offline
- Use endpoint detection and response (EDR) solutions
- Train employees to recognize phishing attempts
- Keep all software and systems updated
- Develop and test an incident response plan

### 2. Phishing and Social Engineering

Phishing attacks have become increasingly sophisticated, using AI-generated content that's nearly indistinguishable from legitimate communications. Attackers target employees through email, SMS, and social media platforms.

**Protection Strategies:**
- Deploy email security gateways with advanced filtering
- Conduct regular security awareness training
- Implement multi-factor authentication (MFA)
- Use email authentication protocols (SPF, DKIM, DMARC)
- Encourage employees to verify suspicious requests through alternative channels

### 3. Supply Chain Attacks

Attackers are targeting software vendors and service providers to gain access to multiple organizations through a single breach. These attacks can affect hundreds or thousands of businesses simultaneously.

**Protection Strategies:**
- Vet third-party vendors thoroughly before integration
- Implement least-privilege access principles
- Monitor third-party access continuously
- Require vendors to meet security standards
- Maintain an inventory of all third-party integrations

### 4. Cloud Security Vulnerabilities

As more businesses migrate to the cloud, misconfigurations and inadequate access controls create significant security gaps. Shared responsibility models can lead to confusion about who's responsible for what.

**Protection Strategies:**
- Regularly audit cloud configurations
- Implement cloud security posture management (CSPM) tools
- Use encryption for data at rest and in transit
- Enable comprehensive logging and monitoring
- Follow cloud security best practices and frameworks

### 5. AI-Powered Attacks

Cybercriminals are using AI to create more convincing phishing emails, generate malware, and automate attacks at scale. AI can also be used to bypass traditional security measures.

**Protection Strategies:**
- Invest in AI-powered security solutions
- Stay informed about emerging AI threats
- Implement behavioral analytics
- Use machine learning for threat detection
- Train security teams on AI attack vectors

### 6. Insider Threats

Whether malicious or accidental, insider threats pose significant risks. Disgruntled employees, compromised accounts, or simple human error can lead to data breaches.

**Protection Strategies:**
- Implement user behavior analytics (UBA)
- Conduct regular access reviews
- Enforce the principle of least privilege
- Monitor and log user activities
- Provide ongoing security training

### 7. IoT Device Vulnerabilities

The proliferation of Internet of Things (IoT) devices creates new attack surfaces. Many IoT devices lack proper security controls and are often overlooked in security strategies.

**Protection Strategies:**
- Maintain an inventory of all IoT devices
- Change default passwords immediately
- Segment IoT devices on separate networks
- Keep firmware updated
- Disable unnecessary features and services

### 8. Zero-Day Exploits

Zero-day vulnerabilities are unknown to vendors and have no patches available. Attackers exploit these vulnerabilities before security patches are developed and deployed.

**Protection Strategies:**
- Implement defense-in-depth strategies
- Use application allowlisting
- Deploy intrusion detection systems
- Monitor threat intelligence feeds
- Have a rapid patching process ready

### 9. Business Email Compromise (BEC)

BEC attacks target employees with access to financial systems, tricking them into transferring funds or revealing sensitive information. These attacks often involve careful research and social engineering.

**Protection Strategies:**
- Implement email authentication protocols
- Require multi-person approval for financial transactions
- Verify payment requests through multiple channels
- Train employees on BEC red flags
- Use email security solutions with BEC detection

### 10. Mobile Device Threats

As mobile devices become primary work tools, they're increasingly targeted by attackers. Mobile malware, unsecured Wi-Fi networks, and device loss create significant risks.

**Protection Strategies:**
- Implement mobile device management (MDM) solutions
- Require device encryption and passcodes
- Use mobile threat defense solutions
- Educate employees on mobile security best practices
- Implement remote wipe capabilities

## Building a Comprehensive Security Strategy

Protecting your business requires a multi-layered approach that combines technology, processes, and people. No single solution can provide complete protection, but a well-designed security strategy can significantly reduce your risk.

### Key Components of a Strong Security Posture

1. **Risk Assessment**: Regularly assess your security posture and identify vulnerabilities
2. **Security Policies**: Develop and enforce comprehensive security policies
3. **Employee Training**: Invest in ongoing security awareness training
4. **Technology Solutions**: Deploy appropriate security tools and technologies
5. **Incident Response**: Prepare for security incidents with tested response plans
6. **Continuous Monitoring**: Implement 24/7 security monitoring and threat detection

## Conclusion

The cybersecurity threat landscape in 2025 demands proactive and comprehensive security measures. By understanding these top threats and implementing the recommended protection strategies, businesses can significantly reduce their risk of falling victim to cyber attacks.

Remember, cybersecurity is not a one-time investment but an ongoing process. Regular assessments, updates, and training are essential to maintain strong security posture. Partnering with experienced cybersecurity professionals can help ensure your business stays protected against evolving threats.

Stay vigilant, stay informed, and prioritize cybersecurity as a fundamental aspect of your business operations.`
  },
  {
    id: 2,
    title: "Why Every Business Needs Cloud Migration in 2025",
    excerpt: "Explore the benefits of cloud computing and why migrating to the cloud is essential for business growth and efficiency.",
    author: "Priya Sharma",
    date: "2025-02-05",
    category: "Cloud Computing",
    readTime: "6 min read",
    image: servicesImages.workspace,
    content: "Full article content goes here..."
  },
  {
    id: 3,
    title: "The Future of AI in Software Development",
    excerpt: "Understand how artificial intelligence is revolutionizing the software development lifecycle and what it means for developers.",
    author: "Amit Patel",
    date: "2025-01-28",
    category: "AI & Technology",
    readTime: "10 min read",
    image: servicesImages.development,
    content: "Full article content goes here..."
  },
  {
    id: 4,
    title: "Building Scalable Web Applications: Best Practices",
    excerpt: "Learn the essential principles and techniques for developing web applications that can scale efficiently.",
    author: "Sneha Reddy",
    date: "2025-01-20",
    category: "Web Development",
    readTime: "7 min read",
    image: servicesImages.team,
    content: "Full article content goes here..."
  },
  {
    id: 5,
    title: "Network Security Essentials for Small Businesses",
    excerpt: "A comprehensive guide to implementing robust network security measures without breaking the bank.",
    author: "Vikram Singh",
    date: "2025-01-15",
    category: "Networking",
    readTime: "9 min read",
    image: servicesImages.general,
    content: "Full article content goes here..."
  },
  {
    id: 6,
    title: "Career Opportunities in IT: Skills You Need in 2025",
    excerpt: "Discover the most in-demand IT skills and how our training programs can help you build a successful tech career.",
    author: "Anjali Verma",
    date: "2025-01-10",
    category: "Career & Training",
    readTime: "5 min read",
    image: trainingImages.session,
    content: "Full article content goes here..."
  }
];

export const testimonials = [
  {
    id: 1,
    name: "Ramesh Gupta",
    company: "CEO, TechStart Pvt Ltd",
    text: "Genius36 Technologies transformed our digital presence with an outstanding website and robust software solution. Their team is professional, responsive, and truly understands business needs.",
    rating: 5,
    image: aboutImages.itProfessionals
  },
  {
    id: 2,
    name: "Dr. Meera Shah",
    company: "Director, Apollo Hospital Bhilai",
    text: "The hospital management system developed by Genius36 has streamlined our operations significantly. Excellent work and great support team.",
    rating: 5,
    image: aboutImages.friendlyTeam
  },
  {
    id: 3,
    name: "Suresh Patel",
    company: "IT Manager, Manufacturing Corp",
    text: "Their networking and infrastructure services are top-notch. They completed our campus-wide network setup ahead of schedule with zero downtime.",
    rating: 5,
    image: aboutImages.teamCollaboration
  }
];

export const stats = [
  { label: "Projects Completed", value: "500+" },
  { label: "Happy Clients", value: "300+" },
  { label: "Years of Experience", value: "10+" },
  { label: "Students Trained", value: "2000+" }
];

export const companyInfo = {
  name: "Genius36 Technologies India LLP",
  tagline: "Your Trusted Pan-India IT Solutions Partner",
  address: "Maroda Tank, Utai Rd, near Police Station, Ajad Nagar, Newai, Bhilai, Chhattisgarh 490006",
  phone: "082005 93901",
  email: "info@genius36tech.com",
  workingHours: "Monday - Saturday: 9:00 AM - 6:30 PM",
  serviceArea: "Bhilai, Chhattisgarh & Pan-India",
  mapUrl: "https://maps.google.com/?q=Maroda+Tank+Utai+Rd+near+Police+Station+Ajad+Nagar+Newai+Bhilai+Chhattisgarh+490006"
};

export const jobOpenings = [
  {
    id: 1,
    title: "Full Stack Web Developer",
    department: "Development",
    location: "Bhilai, Chhattisgarh",
    type: "Full-time",
    experience: "2-4 years",
    description: "We are looking for an experienced Full Stack Developer to join our development team. You will be responsible for developing and maintaining web applications using modern technologies.",
    requirements: [
      "Proficiency in React.js, Node.js, and MongoDB",
      "Experience with RESTful APIs and Git",
      "Strong problem-solving skills",
      "Good communication skills"
    ]
  },
  {
    id: 2,
    title: "UI/UX Designer",
    department: "Design",
    location: "Bhilai, Chhattisgarh",
    type: "Full-time",
    experience: "1-3 years",
    description: "Join our creative team as a UI/UX Designer. You'll design intuitive and engaging user interfaces for web and mobile applications.",
    requirements: [
      "Proficiency in Figma, Adobe XD, or Sketch",
      "Strong portfolio showcasing design skills",
      "Understanding of user-centered design principles",
      "Knowledge of HTML/CSS is a plus"
    ]
  },
  {
    id: 3,
    title: "Network Engineer",
    department: "Infrastructure",
    location: "Bhilai, Chhattisgarh",
    type: "Full-time",
    experience: "3-5 years",
    description: "We need an experienced Network Engineer to design, implement, and maintain our network infrastructure and client networks.",
    requirements: [
      "CCNA/CCNP certification preferred",
      "Experience with LAN/WAN setup and troubleshooting",
      "Knowledge of firewall and security protocols",
      "Strong troubleshooting skills"
    ]
  },
  {
    id: 4,
    title: "IT Trainer",
    department: "Training",
    location: "Bhilai, Chhattisgarh",
    type: "Full-time",
    experience: "2-4 years",
    description: "Looking for an enthusiastic IT Trainer to deliver programming and technology courses to students and professionals.",
    requirements: [
      "Strong knowledge of programming languages (Python, JavaScript, etc.)",
      "Excellent teaching and communication skills",
      "Experience in conducting training sessions",
      "Patience and passion for teaching"
    ]
  },
  {
    id: 5,
    title: "Sales Executive",
    department: "Sales",
    location: "Bhilai, Chhattisgarh",
    type: "Full-time",
    experience: "1-3 years",
    description: "Join our sales team to promote IT services and solutions to businesses. Build relationships and drive business growth.",
    requirements: [
      "Excellent communication and negotiation skills",
      "Basic understanding of IT services",
      "Proven track record in sales",
      "Self-motivated and target-oriented"
    ]
  },
  {
    id: 6,
    title: "DevOps Engineer",
    department: "Development",
    location: "Bhilai, Chhattisgarh",
    type: "Full-time",
    experience: "2-4 years",
    description: "We're seeking a DevOps Engineer to manage our cloud infrastructure, CI/CD pipelines, and deployment processes.",
    requirements: [
      "Experience with AWS, Docker, and Kubernetes",
      "Knowledge of CI/CD tools (Jenkins, GitLab CI)",
      "Scripting skills (Bash, Python)",
      "Understanding of infrastructure as code"
    ]
  }
];
